### PR TITLE
Pil 823: Accessibility CYA routing

### PR DESCRIPTION
**Pull request by alikouhestani**

### CODE REVIEW

File: app/navigation/AmendSubscriptionNavigator.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      .map { nominatedPhoneNumber =>
+        if (nominatedPhoneNumber & userAnswers.get(SubSecondaryCapturePhonePage).isEmpty) {
+          controllers.subscription.manageAccount.routes.SecondaryTelephoneController.onPageLoad()
+        } else {
+          contactDetailCheckYourAnswersRoute
+        }
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  def nextPage(page: Page, mode: Mode = CheckMode, userAnswers: UserAnswers): Call = checkRouteMap(page)(userAnswers)
```
Comment (TevynAllen) [inline app/navigation/AmendSubscriptionNavigator.scala:109]: lets keep public methods at the top of the page and private at the bottom

File: app/navigation/NominatedFilingMemberNavigator.scala [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          controllers.fm.routes.NfmCaptureTelephoneDetailsController.onPageLoad(CheckMode)
+        } else if (userAnswers.get(CheckYourAnswersLogicPage).isDefined) {
+          reviewAndSubmitCheckYourAnswers
+        } else {
+          nfmCheckYourAnswers
+        }
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  def nextPage(page: Page, mode: Mode, userAnswers: UserAnswers): Call = mode match {
```
Comment (TevynAllen) [inline app/navigation/NominatedFilingMemberNavigator.scala:118]: ditto

File: app/navigation/SubscriptionNavigator.scala [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          controllers.subscription.routes.SecondaryTelephoneController.onPageLoad(CheckMode)
+        } else if (userAnswers.finalStatusCheck & userAnswers.get(CheckYourAnswersLogicPage).isDefined) {
+          reviewAndSubmitCheckYourAnswers
+        } else {
+          contactDetailCheckYourAnswersRoute
+        }
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  def nextPage(page: Page, mode: Mode, userAnswers: UserAnswers): Call = mode match {
```
Comment (TevynAllen) [inline app/navigation/SubscriptionNavigator.scala:183]: move to top of the file

File: test/controllers/fm/ContactNfmByTelephoneControllerSpec.scala [hunk 4]
```diff
@@ -98,47 +93,6 @@ class ContactNfmByTelephoneControllerSpec extends SpecBase {
       }
     }
 
-    "redirect to capture telephone page when valid data is submitted with value YES" in {
```
Comment (TevynAllen) [inline test/controllers/fm/ContactNfmByTelephoneControllerSpec.scala:101]: Why did you delete these tests?

File: test/controllers/fm/IsNfmUKBasedControllerSpec.scala [hunk 5]
```diff
@@ -109,53 +102,5 @@ class IsNfmUKBasedControllerSpec extends SpecBase {
       }
     }
 
-    "must redirect to NfmEntityType to choose companyType when Yes is submited" in {
```
Comment (TevynAllen) [inline test/controllers/fm/IsNfmUKBasedControllerSpec.scala:112]: Is this behaviour no longer the case?

File: app/navigation/AmendSubscriptionNavigator.scala (hunk 1 above)
Comment (alikouhestani) [reply app/navigation/AmendSubscriptionNavigator.scala]: Done

File: test/controllers/fm/ContactNfmByTelephoneControllerSpec.scala (hunk 4 above)
Comment (alikouhestani) [reply test/controllers/fm/ContactNfmByTelephoneControllerSpec.scala:101]: Navigation is now done in the Navigator class for each journey and tested it its subsequent spec file

**Review by AKKUM — DISMISSED** [review]

**Review by TevynAllen — APPROVED** [review]
